resource-detection/src/system/hostname.rs

```diff
@@ -1,22 +1,16 @@
 use super::detector::SystemDetectorError;
-#[cfg(unix)]
-use nix::unistd::gethostname;
-use std::ffi::OsString;
 
-/// wrapper for an hostname getter
-#[derive(Default)]
-pub struct HostnameGetter {}
-
-impl HostnameGetter {
-    #[cfg(target_family = "unix")]
-    /// hostname getter
-    pub fn get(&self) -> Result<OsString, SystemDetectorError> {
-        gethostname().map_err(|e| SystemDetectorError::HostnameError(e.to_string()))
-    }
+#[cfg(target_family = "unix")]
+/// hostname getter
+pub fn get_hostname() -> Result<String, SystemDetectorError> {
+    use nix::unistd::gethostname;
+    gethostname()
+        .map_err(|e| SystemDetectorError::HostnameError(e.to_string()))
+        .map(|h| h.into_string().unwrap_or_default())
```

Comment:
It was already like this, but would it make sense to use [to_string_lossy](https://doc.rust-lang.org/nightly/std/ffi/os_str/struct.OsString.html#method.to_string_lossy) instead of `into_string().unwrap_or_default()`?

Reply:
yes!